Implement split datashard status

dahbka-lis opened the pull request:
### Changelog entry

New internal error status for EvWriteResult from data_events: wrong shard state.

...

### Changelog category

* Improvement

### Additional information

...

nepal committed:
Add optimizer flag EmitSkipNullOnPushdown
commit_hash:7615b031b04bf3c1b7ec54e348bf56ad29543c0a

Commit:
Fix checking Bus message limits on reply
commit_hash:f512a7722e094d0c9aefb0ada7750298582b605a

andrei-levitskii committed:
Move clang-format to its own dir

Move clang-format to its own dir

Хотим привозить clang-format как ресурс в тулу для линтинга, но не привозить все clangN
commit_hash:b4d7e2e6953aec15dc41dd523140df5c2ccf685e

georgthegreat committed:
Allow per-module switch to atd_ro_snapshot from arcadia root
commit_hash:d8d2d7bd705c50833ff6e4b03fda17f1c97ca3b7

dimdim1177 committed:
Upgrades for AP in templates, some fixes

Upgrades for AP in templates, some fixes
commit_hash:63f7bbc30e5df3c590e65c9b54346158193efd45

Commit:
[build] ros: Add Python support
commit_hash:72c9a8fb687ae6bec48f141dc2d8d7eb75ed1500

Commit:
Support Android build for libidn

discussion  <HIDDEN_URL>
commit_hash:428d97f442749f3909973ecf3ca24cb9cf1e9f56

Krock21 committed:
Make wire protocol reader validation limits configurable

Currently, WireProtocol**Writer** can write string values of arbitrary size (It does not do validation), but WireProtocol**Reader** validates string lengths while reading, and it uses constant values for dynamic tables for it.

WireProtocol is used to pass results between QueryTracker and engines (YQL, for example), and they do not have to obey dynamic tables limits. I think it is used in RpcProxies too, while working with static tables.

This PR adds ability to configure validation limits for wire protocol reader, so we can configure it in QueryTracker and RpcProxies to work with big values

You can see how it will be used in QueryTracker in this PR: https://github.com/ytsaurus/ytsaurus/pull/944

---

Pull Request resolved: https://github.com/ytsaurus/ytsaurus/pull/943
commit_hash:a03322fcfae53f57c71ba19b12acec307896bb2d

VestniK committed:
Clean ya.make from already removed script
commit_hash:e811732b6c0317af9d138edf53dce95f5f2c63d6

4JustMe4 committed:
Update libcxx to 19 Oct d173ce4a670e88b65c52f6fc1bf10d133ee35704
commit_hash:c1da7a8a6580c15e8af41b1a4847da2163706905

robot-piglet committed:
Intermediate changes
commit_hash:1aefee359cc53471118d5354ffae0495229716df

Commit:
Add tcmalloc soft limit memory handler
commit_hash:30c5264fd43f57937d3355fd8096e0a447ce19e1

olegsidorkin committed:
YT-23122: Use master caches to cache chaos replication card residency
commit_hash:2828610de8fe23e3df32c1c3bcb2a9a754a8d920

robot-piglet committed:
Intermediate changes
commit_hash:a6a45dae782d717c6deaac6a872c24c593927350

robot-piglet committed:
Update contrib/libs/apache/orc to 2.0.3
commit_hash:14f1aaf61c47ab6ad2bd87d8bf69b0fc904cb75b

robot-piglet committed:
Intermediate changes
commit_hash:2ad7a4bc9a9ae76fb2a70feb2681becd0de5c5c7

Commit:
YT-23439: Fix block size estimation in scan writer
commit_hash:6bc02d261d376710e0d08f9372def30d3acfa949

robot-piglet committed:
Intermediate changes
commit_hash:202f8bd6dd019b30f375405881463c996d4546b7

Commit:
Stop reusing error instances
commit_hash:12e3f2d462d1e55b1fb7571acfe28da6d207767d

robot-piglet committed:
Intermediate changes
commit_hash:f2a9fd23d41396b6584b6d133ba2c57fa370863a

alexeevdy committed:
YT-23530: Do not compress response message attachments when it is unneeded
commit_hash:4c7c998726dc08c92babb6c9bd6ab8926c095172

alexeevdy committed:
YT-20603: Change enable_legacy_rpc_codecs's default to false
commit_hash:0a518724f12077d075856c9166f570deb6652bb0

Commit:
YDB Import 622
commit_hash:a2bd81b4ecf17197f540e5096040491a78428687

Commit:
Optimize returning service/method in TServiceContextBase
commit_hash:5bf4c8b492fdc2e24b6225881fb015db596a6c05

Commit:
Cosmetics
commit_hash:e62b6009d92c71c8647ca5915f070fa17e3d9ecf

Commit:
Fix build for vanilla protobuf
commit_hash:8e1de57cc60337fc8ff51a7cbfe9f4efe37d5b98

Commit:
YT-22885: Handle unknown values in (Try)CheckedEnumCast
commit_hash:5ce8019253cdb971d1af36350e3efa3a4ec8545c

robot-piglet committed:
Intermediate changes
commit_hash:892a91804b4df7261c13e779f9cd1a821adf4044

alexeevdy committed:
YT-18537: Expiration of nodes in Sequoia

Sequoia node expiration uses the same mechanism as for master nodes. Master servers are responsible for access tracking and removal of expired nodes  with only difference that removal requests are now executed at Cypress Proxies. Moreover, these changes ensure the proper Sequoia nodes life prolongation via requests.
commit_hash:675c4b5d2ba1ecf158cddb3f2e8dc0bd04d84b72

robot-piglet committed:
Intermediate changes
commit_hash:703034384289d4413f5761736c0a0204198ea13b

Commit:
YT-22885: FromProto -> CheckedEnumCast for enums
commit_hash:6cd1ed86ea24329ed43cda7c28b2c42c55e868c7

Commit:
New version of the tld SKIP_CHECK SKIP_REVIEW
commit_hash:b6c853aa9dfce004915ca07c10eead963195eb97

Commit:
YT-22885: CheckedEnumCast -> FromProto where appropriate
commit_hash:90a3192a5c089821f4301c6dad6d82e65e274cb3

Commit:
Fix batch version of TInvokerWrapper::Invoke
commit_hash:fdb8d24f36e28a431b24b50b7bfd40f6c5a7b001

Commit:
Migrate more addresses to std::string
commit_hash:10f47c96a6a026db591a9ae3270af3e38a27efdd

Commit:
YT-18571: Drop redundant virtual in presence of override
commit_hash:345623486137a7e3c850370cbc99558dc6c20801

Commit:
Use batch invoke to reduce the number of wakeups in TDelayedExecutor
commit_hash:f82fadcde7021c9fcc92f4c9123e0d9d197a79ed

Commit:
Refactor and optimize codicils
commit_hash:30618525152a8c509f47609026d2284a9329534e

Commit:
Automatic release build for ymake, os_ymake

From hash: [00387af4bdd0f4f421ef019e9b2be77851268bf8](HIDDEN_URL
From revision: [15326860](HIDDEN_URL
[CI flow](HIDDEN_URL
Flow triggered by user: [snermolaev](HIDDEN_URL

Update tools: ymake, os_ymake
commit_hash:f1006d685f7cfca5a6e4b6831e3d418f1098438c

Commit:
Fix comment
commit_hash:7686c43f8fb59c7ab99bfcd83ea5108bb7853504

IlyaNiklyaev committed:
PROTO_SCHEMA() module
commit_hash:3c9b927dbb3ddefcaa455eccb51ae1ded847c736

robot-piglet committed:
Intermediate changes
commit_hash:a5695d6abae3e01238ce9328956579759caca454

Commit:
Automatic release build for test_tool, ya_bin, os_ya, os_test_tool

From hash: [5ce8019253cdb971d1af36350e3efa3a4ec8545c](HIDDEN_URL
From revision: [15327756](HIDDEN_URL
[CI flow](HIDDEN_URL
Flow triggered by user: [robot-ci](HIDDEN_URL

Update tools: test_tool, ya_bin, os_ya, os_test_tool
commit_hash:b16a23a4fef3ab23a671ddff7a7cbd4dfc1c21d1

savnadya committed:
YT-22431: Don't use deprecated std::result_of in contrib/libs/apache/arrow
commit_hash:b59cf7dbe69a28e5b49c7def237a8a05fc5cfe4e

E1pp committed:
YT-23435: Parse format string at compile time
commit_hash:804530d1ee861ff42d7d8cad25d9f569b4feaacf

robot-piglet committed:
Intermediate changes
commit_hash:61ff8a8c3d0fbb9a46eb08e98622c938039bc8a9

rvu1024 committed:
YQL-19206 Move contrib/ydb/library/yql/cfg -> yql/essentials/cfg
commit_hash:22adeac140825b580d7674c61994c0404ccf2762

robot-piglet committed:
Intermediate changes
commit_hash:2440e04b3b8bd1cf2e05526e854a12d30f34b8e8

rvu1024 committed:
YQL-19206 Move contrib/ydb/library/yql/tests/s-expressions/suites -> yql/essentials/tests/s-expressions/suites
commit_hash:02beb73b0c072658a9f2dc8bd9762791936b4e29

Commit:
Support JAR_MAIN_CLASS in run.sh - demo

[run.sh](HIDDEN_URL

[run_app.sh](HIDDEN_URL
commit_hash:ad315e3b7e573b557100d239b8913ce6c620fd29

Commit:
cpp/mapreduce: fix style
commit_hash:4b9b636737f162b91295356e0ac17f0ea2beab20

robot-piglet committed:
Intermediate changes
commit_hash:253ff6780bbe06b38b22d1a48de66095f832c01d

robot-piglet committed:
Intermediate changes
commit_hash:6b72cf74512e37b552711d2d0024da75e804e6f1

rvu1024 committed:
YQL-19206 Move contrib/ydb/library/yql/tests/common -> yql/essentials/tests/common
commit_hash:b0bab3353351a5c79f0a64237b103ddba0004fd7

georgthegreat committed:
Enable -Wformat under clang-cl
commit_hash:b2cf3b80a36e4867b03a79f3e2c552f19920016f

dimdim1177 committed:
Fixes for AP semantics

Fixes for AP semantics
commit_hash:aeffb759ee7372e5672d1a7125fbcd5d46d63fe3

Commit:
Update YFM for ya make to latest version

Update YFM for ya make to latest version
commit_hash:54fe358c4864482582ea72f8e2e8eb16815e567b

Commit:
Cosmetics
commit_hash:d90aaaa0da419f9bda8492506c27ff3759b97c23

k-pogorelov committed:
YT-23561: Abort jobs instead of fail on unexpected job phase
commit_hash:a74eea6f0ac2ac4d09210cb34c66305a5baa2f4f

dimdim1177 committed:
Patch annotation processors

Fixes and add sem deps
commit_hash:43b1dd3f53d13a09605153657437d6d8902fcea9

Commit:
use C++ atomic with proper default constructor
commit_hash:1c25da3ce2125ca4d8d1e3e026ef0d2a6d0aa355

Commit:
YT-22943: add context for system errors

* thrown system errors contain context (i.e. host method and request id that produced error)
Type: feature
Component: cpp-sdk
commit_hash:af72a3a37785e9e373e816c2cc072df2076f821d

robot-piglet committed:
Intermediate changes
commit_hash:1bcd4492a3f6a28a2be8e27b5660e98cc8e4c8d0

rvu1024 committed:
YQL-19206 Move contrib/ydb/library/yql/tests/postgresql -> yql/essentials/tests/postgresql
commit_hash:46fdf59714b20cf2b61233a06e58365227d3c8b2

rvu1024 committed:
YQL-19206 Move contrib/ydb/library/yql/tests/sql/suites -> yql/essentials/tests/sql/suites
commit_hash:d0ef1f92b09c94db7c2408f946d2a4c62b603f00

Commit:
YDB Import 623
commit_hash:89b2dfdd1bb0d38b205f23744652b1db522e06c4

robot-piglet committed:
Intermediate changes
commit_hash:5cb62622422884a8fbc7bd1a2d7458b4858d4450

Commit:
YT-20044: Remote copy for dynamic tables with hunk chunks

Support remote copy for dynamic table with hunk chunks.

- Still doesn't work for tables with compression dictionaries.

- Still doesn't work for tables with erasure encoded hunks, mainly due to unsupported remote copy for tables with striped erasure.

Remote copy scheme:

* fetch hunk chunk specs (same as for regular chunks);

* remote copy hunk chunks (TRemoteCopyHunkTask), each job returns HunkChunkIdMapping (old -\> new);

* add HunkChunkIdMapping to remote copy spec to patch chunk meta;

* remote copy chunks (TRemoteCopyTask);

* get hunk chunk list ids for output table;

* attach hunks to chunk lists in which tablets they are used.

Throw an error if the table has compression dictionary config in `@mount_config`.

Throw an error if a static table somehow has a non-empty HunkChunks list.

### Changelog entry

Type: feature

Component: controller-agents, dynamic-tables
commit_hash:bdc959e21c5ea1182ed0ea8268fee3d91e40a427

rvu1024 committed:
YQL-19206 Move contrib/ydb/library/yql/tests/sql/sql2yql -> yql/essentials/tests/sql/sql2yql
commit_hash:137ab446f8ed4df274d02b251cbe5e2bb08c8194

andrey-khropov committed:
Add a clarifying comment about exceptions in TryFromString* functions
commit_hash:60b664d99f74a0d1ec00e0ab843e87583a5360a4

dimdim1177 committed:
Fix javac flags, full support of RUN_JAVA_PROGRAM

Fix javac flags, full support of RUN_JAVA_PROGRAM
commit_hash:0d9cc80fef351cb6c25839b90c34483448ecbe7c

dimdim1177 committed:
Style fixes

Style fixes
commit_hash:3a65d76039bb3c84b47e849c0ca9833d9b6d6e7a

Nechda committed:
Add suppressions for ubsan in openssl
commit_hash:54d8848f6168c0370b9bcbd164290fc4fc1c2128

robot-piglet committed:
Intermediate changes
commit_hash:032737595139b2cf9c9ffdb1c8fd20846852fc6d

Commit:
New version of the tld SKIP_CHECK SKIP_REVIEW
commit_hash:4591e7a8126c65672359f0cdee64ad789ca9ae74

Commit:
/util: streaming CityHash
commit_hash:bd4032fbf4c4ece089ad071747194b38df6c5edd

andrei-levitskii committed:
Use clang-format resource in style cpp linter

Use clang-format resouce in style cpp linter
commit_hash:440dce939163f75f5652a9d119a2d5f064e47e6e

robot-piglet committed:
Intermediate changes
commit_hash:c2908947ab85dde6631ac29f6d9fbf43bbb2648d

Commit:
Part of PR. Style

Часть большого ПР REVIEW:7264088
commit_hash:0f5b03fbbed0ac30f734943309e3ef5cd4d7a30e

rvu1024 committed:
YQL-19206 Move contrib/ydb/library/yql/tests/sql/yt_native_file -> yt/yql/tests/sql/yt_file
commit_hash:0c9c4ff8a4fbff1f8ad41fc95097d56401249616

alexv-smirnov committed:
Import libraries 241120-1113

rvu1024 committed:
Port changes from arcadia (#11796)

shadchin committed:
Update flake8 binaries
commit_hash:33644a2ad5fc56ac738869b46aa5d5e991c8ef9a

Commit:
Update YFM for ya make to latest version

Update YFM for ya make to latest version
commit_hash:9621b73ea72f6a48e9fdb3612cef01e5eddad28f

rvu1024 committed:
Fix dq/hybrid tests (#11808)

rvu1024 committed:
Fix YQL_UDF_YDB_TEST macro
commit_hash:231818a85581b4c119ea310cb8ce8c47934f327b

Commit:
YT-23461: Support different columns in meta

* Changelog entry
  Type: fix
  Component: proxy

Add the ability to read tables with different number of columns in chunk meta in arrow format.
commit_hash:d8d654f4addbeecab2529033ddff68f31521f104

kobzonega committed:
breakpad firstly extract backtrace from current thread (#11815)

gridnevvvit committed:
Fix overflow case in stream index lookup join (#11818)

robot-piglet committed:
Intermediate changes
commit_hash:c9890b6966b6c4d2d0cb13bda9b30004d232450e

Commit:
Added cluster field to TRichYPath

Add cluster field to TRichYPath

### Changelog entry
Type: feature
Component: cpp-sdk
commit_hash:eff6ccc4add04d64520c427c194424913ea2dfc4

dimdim1177 committed:
Fix most Gradle proto builds

Fix most Gradle proto builds
commit_hash:d601ed32e402ea33de324070c013643e09f2b1c1

eivanov89 committed:
Use tail sends when replying from QP (#11813)

robot-piglet committed:
Intermediate changes
commit_hash:9085ddac9f80e60b5b938027d444ed98e80ef95a

Hor911 committed:
Sync EvaluteExpr execution (#11801)

fexolm committed:
Sensor for number of shards within single BulkUpsert data (#11786)

georgthegreat committed:
contrib/libs/poco: Backport PR #2943 to fix -Wimplicit-const-int-float-conversion

Backport [PR #2943](https://github.com/pocoproject/poco/pull/2943) from poco upstream.
commit_hash:5013b65cd3f7e719f6489c889c710eea3ad91537

ivanmorozov333 committed:
Accessors memory limit on background (#11816)

andrei-levitskii committed:
Migrate to STYLE_CPP (#11523)

Commit:
Column Family for ColumnTable (#9657)

Commit:
Extend TTL syntax to support tiers

RFC: **[nda.ya.ru/t/JsIT3hp679nYxn](https://nda.ya.ru/t/JsIT3hp679nYxn)**
commit_hash:a0a4f65b24ee591cb76fd3cf253ffe24a01bfaf5

robot-piglet committed:
Update contrib/python/types-protobuf to 5.28.0.20240924
commit_hash:0a209dcd80a5c303e9cbc293056b422eb96242a1

apollo1321 committed:
Fix issues in rXXXXXX

Fix issues
commit_hash:da56a7560d342885bc31423bcc6dd53188cc4841

alexv-smirnov committed:
build: refresh Embedded UI (v6.36.1) (#11833)

Co-authored-by: Raubzeug <Raubzeug@users.noreply.github.com>

lll-phill-lll committed:
Fix udfs path in tests runner (#11821)

robot-piglet committed:
Intermediate changes
commit_hash:3d8c36a9891955ab700f018337e599a271197819

Commit:
YT-23542: Add local throttlers to distributed throttler member attributes.
commit_hash:786aac617c6a994ba9f2f527bb6c64c55f48be29

kardymonds committed:
YQ-3888 Fix "session already exists" error (#11826)

snaury committed:
Make sure mailboxes in test actorsystems are in a locked state (#11837)

nikvas0 committed:
Stock bench for Olap shards (#11757)

GrigoriyPA committed:
YQ-3892 fix data race in json filters (#11827)

SammyVimes committed:
Fix blobstorage crypto module build for aarch64 (#11840)

ponasenko-rs committed:
YT-23585: Fix race between Cancel and GetCanceledError in Canceler
commit_hash:2c1b31d95037975cf4703cb90f81232f880a37e6

ivanmorozov333 committed:
fix filter usage for partial reading (#11835)

GrigoriyPA committed:
YQ-3890 added per node partition limit (#11830)

Commit:
[arc][getopt]fix enforcment of RequiredArgument().DisableSpaceParse()
commit_hash:9c2ada5b95be3f810d2750406d74d839bd2598ff

iddqdex committed:
increase timeout on tpch 1000 (#11850)

kardymonds committed:
YQ-3873 Add query id to error /  fix clang14 build (#11846)

naspirato committed:
unmute stable tests (#11856)

robot-piglet committed:
Intermediate changes
commit_hash:50996f174ea1b659bc41e9f4e9f27328060af2ca

Commit:
Update YFM for ya make to latest version

Update YFM for ya make to latest version
commit_hash:de0c6142717a3513a711df08540adc3ba5113307

dimdim1177 committed:
Fix proto with proto-libraries, refactor errorprone

Refactor errorprone

Fix building protos in Gradle
commit_hash:54d1cdf9df00e7ab30b991ddf1f534ab5992ef3c

alexvru committed:
Fix blob recovery actor bug (#11857)

Commit:
YDB FQ: support Logging as an external data source (#11765)

nepal committed:
Fix filter pushdown over Aggregate with session/hopping window
commit_hash:3d4989f7a92e9b330cd4b19f1850cdca544a1d64

loochek committed:
ListSample/ListSampleN/ListShuffle implementation
commit_hash:987b10b398caa89eee8b94b33f9ea1dc74197223

iddqdex committed:
use yaml for table description instead yql (#11811)

georgthegreat committed:
Enable -Wreturn-std-move
commit_hash:c7f0699a8f585dc8140bded7a65b017808bebb38

robot-brewer committed:
Release clang-format-18 #2

https://github.com/yandex/toolchain-registry/releases/tag/clang-format-18-v2
commit_hash:0cc68e232a14adf66a88617e9b1f7f18facd2dce

shnikd committed:
Support ALTER SEQUENCE command

<https://github.com/ydb-platform/ydb/issues/11763>
commit_hash:5611cbe2f0e09543749e9866c5db8d427649e754

robot-piglet committed:
Update contrib/libs/expat to 2.6.4
commit_hash:9a71f6b42bb4dd7cf1281e4b72c5da4a3c18a399

andrey-khropov committed:
Remove TRbTree and TAvlTree from forward declarations as they are not in util
commit_hash:7f3c30234700758cc6218bdef0fa2e6862f2c32e

Commit:
Public TValuePackerTransport.IsBlock_ field
commit_hash:1460634e062682d017761c493ba36122257278e9

robot-piglet committed:
Update contrib/python/types-protobuf to 5.28.3.20241030
commit_hash:34dfb9be1465bd259bf2f5c90a3114542cd87d6d

Commit:
nots/*: Стабилизация trunk

Суть изменений:
1. Наполнение папки ~/.nots/nm_store/$MODDIR/node_modules прямо в builder (перед установкой модулей в $BINDIR)
2. Это все равно нужно было бы делать в nots/cli, а теперь оно наполняется заранее, что обеспечивает структуру node_modules выше по уровню, чем virtual-store (важно для кастомных резолверов пакетов, ищущих пакеты выше по иерархии)
3. Убран предварительный запуск `pnpm install`, пакеты качаются один раз (тарболы в ya make)
4. После `ya make` все равно запускается `pnpm install` для гарантии (там еще всякие действия делаются)

# Всем, кого робот призовет в PR
## Пожалуйста, игнорируйте, это массовая проверка, в транк не попадут изменения в ваших проектах
## Нам нужно лишь стриггерить CI, чтобы проверить, что никто не сломался. Правки в pnpm-lock.yaml файлах будет откачены.
commit_hash:d124ecb77bda10bd8e975078382c685ac35c8928

nikvas0 commented on ydb/core/kqp/runtime/kqp_write_actor.cpp:
```diff
@@ -535,24 +535,28 @@ class TKqpTableWriteActor : public TActorBootstrapped<TKqpTableWriteActor> {
             return;
         }
+        case NKikimrDataEvents::TEvWriteResult::STATUS_WRONG_SHARD_STATE: {
+            CA_LOG_E("Got INTERNAL ERROR for table `"
```
WRONG SHARD STATE instead of INTERNAL ERROR here

nikvas0 commented on ydb/core/kqp/runtime/kqp_write_actor.cpp:
```diff
@@ -535,24 +535,28 @@ class TKqpTableWriteActor : public TActorBootstrapped<TKqpTableWriteActor> {
+                    << " ShardID=" << ev->Get()->Record.GetOrigin() << ","
+                    << " Sink=" << this->SelfId() << "."
+                    << getIssues().ToOneLineString());
+            ResetShardRetries(ev->Get()->Record.GetOrigin(), ev->Cookie);
```
Looks like there was a bug)
We need RuntimeError in case of !InconsistentTx

robot-piglet committed:
Intermediate changes
commit_hash:e8f03143f071def87f56790176fbdc918784ed35

Commit:
Signed Enumerate
commit_hash:af73a3ea146e2eb6141cb19913e534d75ba573fd

Enjection committed:
Add fallback to ssl port on ydb driver init (#11869)

alexv-smirnov committed:
Import libraries 241122-0919

pnv1 committed:
Improve csv import, step 3 (#11864)

swalrus1 committed:
add subcodes to tx-proxy reply code counters (#11862)

omgronny committed:
YT-22455: Introduce list jobs continuation token

[nodiff:caesar]
commit_hash:d45b3da99e7b19120e02298ca6e87c02cc800ea2

Enjection committed:
Extract schemeshard protos p.1 (#11803)

pixcc committed:
Support configuring scale recommender policies in console (#11798)

Commit:
YDB Import 625
commit_hash:bbf24592c367fa158fbc53e041fe367374d0750e

Commit:
YT_DEFINE_GLOBAL for profilers
commit_hash:c9f0f790039ceeef149366a14c9063025fbb1e0c

Hor911 committed:
Correct Rows count in MultiType Input Channels (#11859)

Commit:
Disable Y_ENSURE() for improper invariant

Правильный фикс здесь: <https://github.com/ydb-platform/ydb/pull/11129/files>

Ему неповезло попасть на переезд yql в аркадию, поэтому ждём его окончания.

Y_ENSURE() вызывает падения в проде - до этого он был Y_ASSERT(), но так как проверяемый инвариант не всегда выполняется, то просто убрал.
commit_hash:76966d1e5c858de4e099beccd34c9570dd9f6883

Commit:
ya style clang format. Braces style

Принудительно добавлять скобки для однострочников при использовании ya style.

чтобы такого не было
```cpp
        for (const TString& sliceName : sliceNames)
            for (auto& test : allWebRankingSlices)
                if (sliceName == ToString(test.first))
                    return test.second;
        return algo2Domain;
```

Зависит от
- REVIEW:7273779
- REVIEW:7273927
- REVIEW:7274020
- REVIEW:7274173
- REVIEW:7274863
- REVIEW:7274942
- REVIEW:7286577
- REVIEW:7286432
- REVIEW:7286620
- REVIEW:7308852
commit_hash:44ce41c86610dfd52cc9a1656ff1da6b4e411d6b

kuzmich321 committed:
tempo fix for win (pybuild)

Предлагаю пока так отрубить для винды. Хочется, чтобы покрытие работало
commit_hash:9b3ec4a246f6597795dc0b2d12d6524bb688e129

azevaykin committed:
Remove old restriction: keys with Uint8 column values >127 are currently prohibited (#11886)

alexvru committed:
Revert "Support arbitrary chain set in huge blob keeper heap (#11566)" (#11891)

kardymonds committed:
YQ-3900 Fix ScheduleHeartbeat() if already connected (#11888)

robot-piglet committed:
Intermediate changes
commit_hash:d0db92345dc96a43a16424238c1eb2812040a13d

nshestakov committed:
Fix URL for transaction page of PQ tablet (#11900)

robot-piglet committed:
Intermediate changes
commit_hash:d3ad801a6e5fb7eb38c6c78286f53c3b165b87c2

Commit:
YT-23422: do not write command line into attributes of operations

* Changelog entry
Type: feature
Component: cpp-sdk

C++ SDK doesn't write launching process command line to operation attributes.
commit_hash:c50cbc2a31f5a3b733833767fe94019a0d7615d3

Commit:
fix buffered stream interleaving for proto format
commit_hash:66e6b75f97ea253ddbf4c97a180f9f971c59f93b

georgthegreat committed:
Unify -Wno arguments generation
commit_hash:33157feb41c65ab08e797adea3b0be5ed6e5bc3f

alephonea committed:
Change priority for rule NonOptimalTableContent (#11625)

Hor911 committed:
Correct Rows count in Block Output Channels (#11893)

rvu1024 committed:
Check custom error in xfail tests. Remove canonization from dq/hybrid tests

1. xfail тесты теперь должны содержать спец комментарий `custom error` с ожидаемым текстом ошибки

2. Все suites для xfail размечены спец. коментариями, некоторые тесты разбиты на несколько

3. Включены тесты с Python и Javascript udf для yt провайдера и переканонизированы (были отключены в github)

4. Для тяжелых xfail тестов аналогично убрана канонизация и включена проверка по спец. коментарию

5. Из dq_file/hybrid тестов убрана канонизация. Осталась только кросс-проверка результатов с yqlrun и блочным режимом

6. В dqrun теперь используется exception_policy вместо fail_policy (для корректной выдачи ошибок в тестах)
commit_hash:a92b3835f86b0c01225e81e3f28bb6d11d8d67a3

alexv-smirnov committed:
Import libraries 241123-2111

rvu1024 committed:
Check custom error in xfail tests. Remove canonization from dq/hybrid tests (#11910)

rvu1024 committed:
YQL-19206 Fix YQL_PYTHON* macros
commit_hash:efcdd5f8a76a33ea53572df5a214f2de61c7775f

rvu1024 committed:
Change YQL_BASE_DIR

Commit:
Update docs for datetime udf (#11746)

alexvru committed:
Remove REAL infix from huge blob size (#11916)

iddqdex committed:
Set position to state only if all data portions succesfully send (#11912)

iddqdex committed:
Fix and simplify logic with multiprocess TPC-* data generation (#11913)

naspirato committed:
Unmute stable tests (#11924)

ivanmorozov333 committed:
validate dangerouse construction (#11873)

maximyurchuk committed:
Accept True in stability test (#11932)

ivanmorozov333 committed:
fix empty variable usage (#11926)

pixcc committed:
Update ydbops examples for CMS docs (#11793)

vitstn committed:
Drop yql fastcheck before move (#11937)
